From 8d11cff135065494156414ca6c5921339e2639cb Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Wed, 26 Oct 2022 18:10:46 -0400
Subject: [PATCH 2/3] BpfMap() implemented new checks for kernel >=4.14 but as
 assert. Since we removed the asset but moved those checks into isOk, use isOk
 as well

Change-Id: Ia7fa7494a2c320aa521e915d4926e8748d472689
---
 service-t/native/libs/libnetworkstats/BpfNetworkStats.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/service-t/native/libs/libnetworkstats/BpfNetworkStats.cpp b/service-t/native/libs/libnetworkstats/BpfNetworkStats.cpp
index 9ebef4d6d2..ee30f3db30 100644
--- a/service-t/native/libs/libnetworkstats/BpfNetworkStats.cpp
+++ b/service-t/native/libs/libnetworkstats/BpfNetworkStats.cpp
@@ -102,13 +102,13 @@ int bpfGetIfaceStatsInternal(const char* iface, Stats* stats,
 int bpfGetIfaceStats(const char* iface, Stats* stats) {
     BpfMapRO<uint32_t, StatsValue> ifaceStatsMap(IFACE_STATS_MAP_PATH);
     int ret;
-    if (!ifaceStatsMap.isValid()) {
+    if (!ifaceStatsMap.isValid() || !ifaceStatsMap.isOk()) {
         ret = -errno;
         ALOGE("get ifaceStats map fd failed: %s", strerror(errno));
         return ret;
     }
     BpfMapRO<uint32_t, IfaceValue> ifaceIndexNameMap(IFACE_INDEX_NAME_MAP_PATH);
-    if (!ifaceIndexNameMap.isValid()) {
+    if (!ifaceIndexNameMap.isValid() || !ifaceStatsMap.isOk()) {
         ret = -errno;
         ALOGE("get ifaceIndexName map fd failed: %s", strerror(errno));
         return ret;
-- 
2.25.1

